recipes/fcl/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class FclConan(ConanFile):
+    name = "fcl"
+    description = "C++11 library for performing three types of proximity " \
+                  "queries on a pair of geometric models composed of triangles."
+    license = "BSD-3-Clause"
+    topics = ("conan", "fcl", "geometry", "collision")
+    homepage = "https://github.com/flexible-collision-library/fcl"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_octomap": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_octomap": True
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
```

Comment:
Is the problem Windows or MSVC specific? If mingw causes problems too, I would replace the test with `self.settings.os == "Windows"`

Reply:
After some tests, it doesn't fail to build with MinGW and shared (at least with Release build type), but test_package fails.
I've droped support of shared on Windows, but also of Visual Studio < 2017.